src/components/Input.jsx

```diff
@@ -97,4 +102,70 @@ class PasswordInputComponent extends Component {
   }
 }
 
+const PasswordStrength = props => {
+  const { password, className, ...rest } = props
+  const strength = passwordHelper.getStrength(password)
+
+  return (
+    <progress
+      step="1"
+      min="0"
+      max="100"
+      value={strength.percentage}
+      className={cx(styles[`pw-${strength.label}`], className)}
+      {...rest}
+    />
+  )
+}
+
 export const PasswordInput = translate()(PasswordInputComponent)
+
+const HideShowButton = props => {
+  const { hidden, ...rest } = props
+
+  return (
+    <button
+      type="button"
+      style={{
+        height: '100%',
+        width: 48,
+        backgroundColor: 'transparent',
+        border: '0'
+      }}
+      {...rest}
+    >
+      <Icon
+        icon={hidden ? 'eye' : 'eye-closed'}
+        size={16}
+        color="var(--coolGrey)"
+      />
+    </button>
+  )
+}
+export const NewPasswordInput = props => {
+  const { className, showStrength, error, ...rest } = props
+  const [hidden, setHidden] = useState(true)
+
+  return (
+    <div className={cx(styles['coz-pwd-input'], className)}>
+      <InputGroup
+        append={
+          <HideShowButton hidden={hidden} onClick={() => setHidden(!hidden)} />
+        }
+        className={cx(
+          styles['coz-input-group'],
```

Comment:
Nit: This styles removes the border of the `input`, I think it would be best if this behavior was coded inside InputGroup.

Reply:
This has been extracted to cozy-ui (https://github.com/cozy/cozy-ui/pull/1311). And in https://github.com/cozy/cozy-settings/pull/269/commits/8f8ce042ce206daca64194c0bfc726c615a7abca I used `Input` for cozy-ui, so I removed stylus code that added borders to all inputs (and required to manually reset the borders in some contexts)